common/org.testeditor.dsl.common/src/org/testeditor/dsl/common/ide/util/ProjectContentGenerator.xtend

```diff
@@ -0,0 +1,381 @@
+/*******************************************************************************
+ * Copyright (c) 2012 - 2016 Signal Iduna Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors:
+ * Signal Iduna Corporation - initial API and implementation
+ * akquinet AG
+ * itemis AG
+ *******************************************************************************/
+package org.testeditor.dsl.common.ide.util
+
+import java.io.ByteArrayInputStream
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.m2e.core.MavenPlugin
+import org.eclipse.m2e.core.project.ResolverConfiguration
+
+/**
+ * Generator to genrate content to a new test project
+ */
+class ProjectContentGenerator {
+
+	public static val MAVEN = "Maven"
+	public static val GRADLE = "Gradle"
+	public static val WEBFIXTURE = "Web Fixture"
+	// NOT API yet.
+	static val SWINGFIXTURE = "Swing Fixture"
+
+	def createProjectContent(IProject project, String[] fixtures, String buildsystem, boolean demo,
+		IProgressMonitor monitor) throws CoreException{
+		if (buildsystem == MAVEN) {
+			var IFile buildFile = project.getFile("pom.xml")
+			buildFile.create(new ByteArrayInputStream(getPomContent(fixtures, project.name).getBytes()), IResource.NONE,
```

Comment:
`IFile.create` is called several times here. I would prefer to first write the contents to a variable and then call it.
    
    val pomContents = getPomContents(fixtures, project.name)
    buildFile.create(new StringInputStream(pomContents), ...)